### PR TITLE
Remove backend argument of init_parallel_env

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Function optimization
### PR changes

APIs
### Describe

Remove backend argument of init_parallel_env, reason: 

image

related PR: https://github.com/PaddlePaddle/Paddle/pull/26044

### CODE REVIEW

File: python/paddle/distributed/parallel.py [hunk 1]
```diff
@@ -112,30 +103,28 @@ def _check_var_exists(var_name):
...
+    strategy.current_endpoint = ParallelEnv().current_endpoint
+    if strategy.nranks < 2:
+        return
+    # NOTE(chenweihang): [ why config global place here? ]
+    # the dygraph mode will be set to default mode, 
+    # users will not call `dygraph.guard` or `enable_dygraph`
+    # directly, if they want to switch default place,
+    # they need to call a function to change default place,
+    # here just set correctly place to users
+    place = core.CUDAPlace(ParallelEnv().device_id)
```
Comment (guru4elephant) [inline python/paddle/distributed/parallel.py:129]: Can I call this function with CPU whl?

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/distributed/parallel.py (hunk 1 above)
Comment (chenwhql) [reply python/paddle/distributed/parallel.py:129]: thx, add cuda check for this function

**Review by guru4elephant — DISMISSED** [review]
LGTM, I think init_parallel_env() can be used in CPU whl, the difference is we init gloo related functions. For GPU whl, we'd better init both nccl and gloo for compatibility of cpu/gpu collective operators. @sandyhouse @chenwhql please improve this function in the following updates.

**Comment by chenwhql** [comment]
> LGTM, I think init_parallel_env() can be used in CPU whl, the difference is we init gloo related functions. For GPU whl, we'd better init both nccl and gloo for compatibility of cpu/gpu collective operators. @sandyhouse @chenwhql please improve this function in the following updates.

thx, of course, we should support init CPU parallel env in this function later

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by kolinwei — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**